store.go

```diff
@@ -232,6 +246,15 @@ func (s *Store) markReady() {
 	}
 }
 
+// Demote instructs store to destroy its primary lease, if any.
+// Store will wait momentarily before attempting to become primary again.
+func (s *Store) Demote() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	close(s.demoteCh)
+	s.demoteCh = make(chan struct{})
+}
+
```

Comment:
This function is to simulate losing primary status without a node actually going down.